evm-fungible-token/tests/general.rs

```diff
@@ -0,0 +1,111 @@
+}
+
+fn init() -> (
+    UserAccount,
+    ContractAccount<EthConnectorContract>,
+    UserAccount,
+) {
+    let master_account = init_simulator(None);
+    let eth_ecc = "79183fdbd80e2d8AeA1aCaA2f67bFb8a36d40A80";
+    let initial_balance = to_yocto("100000");
+
+    let contract_account = deploy! {
+        contract: EthConnectorContract,
+        contract_id: "contract",
+        bytes: &TOKEN_WASM_BYTES,
+        signer_account: master_account,
+        init_method: new("contract".into(), eth_ecc.into())
+    };
+    let alice = master_account.create_user("alice".to_string(), initial_balance);
+    (master_account, contract_account, alice)
+}
+
+#[test]
+fn init_test() {
+    let (_master_account, _contract_account, _alice) = init();
+}
+
+#[test]
+fn test_sim_deposit() {
+    let (master_account, contract, _) = init();
+    /*
+        // Deploy Eth Verifier Contract
+        let status_id = "acc1".to_string();
+        let status_amt = to_yocto("100");
+        let res = call!(
+            master_account,
+            contract.deploy(status_id.clone(), status_amt.into()),
+            STORAGE_AMOUNT * 1000,
+            DEFAULT_GAS
+        );
+        println!("CALL: {:#?}", res);
+        println!("#2: {:#?}", res.promise_results());
+    */
+    let proof = Proof {
+        log_index: 0,
+        log_entry_data: vec![],
+        receipt_index: 0,
+        receipt_data: vec![],
+        header_data: vec![],
+        proof: vec![],
+    };
+    let _res = call!(
+        master_account,
+        contract.deposit(proof.clone()),
+        gas = DEFAULT_GAS * 3
+    );
+
+    // println!("#1: {:#?}", _res.promise_results());
+
+    let acc_id = ValidAccountId::try_from("rcv1").unwrap();
+    let res = view!(contract.ft_balance_of(acc_id));
+    let minted_balance = res.unwrap_json::<U128>();
+    assert_eq!(minted_balance, U128::from(100));
+}
+
+#[test]
+fn test_sim_withdraw() {
+    let (master_account, contract, _alice) = init();
+    let proof = Proof {
+        log_index: 0,
+        log_entry_data: vec![],
+        receipt_index: 0,
+        receipt_data: vec![],
+        header_data: vec![],
+        proof: vec![],
+    };
+    call!(
+        master_account,
+        contract.deposit(proof.clone()),
+        gas = DEFAULT_GAS * 3
+    );
+    let acc_id = ValidAccountId::try_from("rcv1").unwrap();
+    let res = view!(contract.ft_balance_of(acc_id.clone()));
+    let minted_balance = res.unwrap_json::<U128>();
+    assert_eq!(minted_balance, U128::from(100));
+
+    let mut proof1 = proof.clone();
+    proof1.log_index = 1;
+
+    let _res = call!(
+        master_account,
+        contract.withdraw(proof1.clone()),
+        gas = DEFAULT_GAS * 3
+    );
+
+    // println!("#1: {:#?}", _res.promise_results());
+    let res = view!(contract.ft_balance_of(acc_id));
+    let minted_balance = res.unwrap_json::<U128>();
+    assert_eq!(minted_balance, U128::from(95));
+}
```

Comment:
Also test fail with invalid proof, and fail with reuse proof?

Reply:
> Also test fail with invalid proof, and fail with reuse proof?

For now, we are testing with proof data for `eth-prover`. So testing still in progress, because we "bridging" all that things. If you mean `Proof` dependency failure - it was fixed with lastest commit.